### PR TITLE
Avoid this escape.

### PR DESCRIPTION
**Pull request by baldersheim**
@vekterli PR

### CODE REVIEW

File: documentapi/src/main/java/com/yahoo/documentapi/ProgressToken.java [hunk 1]
```diff
@@ -75,59 +75,49 @@ public void setState(BucketState state) {
...
-                // MSBs differ
-                return ((key >>> 63) > (other.key >>> 63)) ? 1 : -1;
-            }
-            // Mask off MSBs since we've already checked them, and with MSB != 1
-            // we know the ordering will be consistent
-            if ((key & 0x7FFFFFFFFFFFFFFFL) < (other.key & 0x7FFFFFFFFFFFFFFFL)) {
-                return -1;
-            } else if ((key & 0x7FFFFFFFFFFFFFFFL) > (other.key & 0x7FFFFFFFFFFFFFFFL)) {
-                return 1;
+                if ((key & 0x8000000000000000L) != (other.key & 0x8000000000000000L)) {
```
Comment (vekterli) [inline documentapi/src/main/java/com/yahoo/documentapi/ProgressToken.java:86]: Unintentional added indent?

**Review by vekterli — COMMENTED** [review]
My main concern is that all of these classes are contained in `@PublicAPI` packages, so removing or `final`-izing classes (top-level classes in particular) might be a breaking change. Other than that LGTM.

**Review by vekterli — APPROVED** [review]
:+1:

File: documentapi/src/main/java/com/yahoo/documentapi/ProgressToken.java (hunk 1 above)
Comment (baldersheim) [reply documentapi/src/main/java/com/yahoo/documentapi/ProgressToken.java]: fixed

**Pull request merged**